parallel_algorithms/P3179/P3179.md

```diff
@@ -0,0 +1,6630 @@
+         class Proj1 = identity, class Proj2 = identity>
+  requires indirectly_copyable<iterator_t<R1>, iterator_t<R2>> &&
+           sortable<iterator_t<R2>, Comp, Proj2> &&
+           indirect_strict_weak_order<Comp, projected<iterator_t<R1>, Proj1>,
+                                      projected<iterator_t<R2>, Proj2>>
+  ranges::partial_sort_copy_result<borrowed_iterator_t<R1>, borrowed_iterator_t<R2>>
+    ranges::partial_sort_copy(Ep&& exec, R1&& r, R2&& result_r, Comp comp = {},
+                              Proj1 proj1 = {}, Proj2 proj2 = {});
+```
+:::
+
+[1]{.pnum} Let *`N`* be `min(last - first,  result_last - result_first)`. Let `comp` be `less{}`, and `proj1` and `proj2`
+be `identity{}` for the overloads with no parameters by those names.
+
+[2]{.pnum} *Mandates*: For the overloads in namespace `std`, the expression `*first` is writable
+([iterator.requirements.general]{- .sref}) to `result_first`.
+
+[3]{.pnum} *Preconditions*: For the overloads in namespace `std`, `RandomAccessIterator` meets the *Cpp17ValueSwappable*
+requirements ([swappable.requirements]{- .sref}), the type of `*result_first` meets the *Cpp17MoveConstructible* (Table 31)
+and *Cpp17MoveAssignable* (Table 33) requirements.
+
+[4]{.pnum} For iterators `a1` and `b1` in `[first, last)`, and iterators `x2` and `y2` in `[result_first, result_last)`,
+after evaluating the assignment `*y2 = *b1`, let *`E`* be the value of
+
+`bool(invoke(comp, invoke(proj1, *a1), invoke(proj2, *y2)))`.
+
+Then, after evaluating the assignment `*x2 = *a1`, *`E`* is equal to
+
+`bool(invoke(comp, invoke(proj2, *x2), invoke(proj2, *y2)))`.
+
+[*Note 1*: Writing a value from the input range into the output range does not affect how it is ordered by `comp` and
+`proj1` or `proj2`. — *end note*]
+
+[5]{.pnum} *Effects*: Places the first *`N`* elements as sorted with respect to `comp` and `proj2` into the range
+`[result_first, result_first + @_N_@)`.
+
+[6]{.pnum} *Returns*:
+
+- [6.1]{.pnum} `result_first + @_N_@` for the overloads in namespace `std`.
+- [6.2]{.pnum} `{last, result_first + @_N_@}` for the overloads in namespace `ranges`.
+
+[7]{.pnum} *Complexity*: Approximately `(last - first) * log@_N_@` comparisons, and twice as many projections.
+
+## Modify `is_sorted` in [is.sorted]{- .sref} ## {#modify_is_sorted}
+
+```cpp
+template<forward_iterator I, sentinel_for<I> S, class Proj = identity,
+         indirect_strict_weak_order<projected<I, Proj>> Comp = ranges::less>
+  constexpr bool ranges::is_sorted(I first, S last, Comp comp = {}, Proj proj = {});
+template<forward_range R, class Proj = identity,
+         indirect_strict_weak_order<projected<iterator_t<R>, Proj>> Comp = ranges::less>
+  constexpr bool ranges::is_sorted(R&& r, Comp comp = {}, Proj proj = {});
+```
+*Effects*: Equivalent to: `return ranges​::​is_sorted_until(first, last, comp, proj) == last;`
+
+::: add
+```cpp
+template<@_execution-policy_@ Ep, random_access_iterator I, sized_sentinel_for<I> S, class Pr
```

Comment:
You use `OR` here, so `OS` for the sentinel in the algorithms would be consistent.

Reply:
In most cases we use OutR though.

I guess it makes sense for us to agree on the use of OutR/OutS vs. OR/OS vs. something else. I do not mind much, but slightly prefer the more verbose version just because of potential confusion that OR and, to lesser degree, OS might cause.

Reply:
I have added this to "LWG questions" section as mentioned in the response to one of the previous comments.